modulus/distributed/utils.py

```diff
@@ -22,6 +22,26 @@
 from .manager import DistributedManager
 
 
+def compute_split_shapes(size: int, num_chunks: int) -> List[int]:
+
+    # treat trivial case first
+    if num_chunks == 1:
+        return [size]
+
+    # first, check if we can split using div-up to balance the load:
+    chunk_size = (size + num_chunks - 1) // num_chunks
+    last_chunk_size = max(0, size - chunk_size * (num_chunks - 1))
+    if last_chunk_size == 0:
+        # in this case, the last shard would be empty, split with floor instead:
+        chunk_size = size // num_chunks
+        last_chunk_size = size - chunk_size * (num_chunks - 1)
+
+    # generate sections list
+    sections = [chunk_size for _ in range(num_chunks - 1)] + [last_chunk_size]
```

Comment:
Yes, one can do round robin distribution. However, I don't see that there is a huge imbalance here, since this is a case which rarely gets hit I hope. We can change to assigning the remainder round robin though, I don't mind. What do you prefer?

Reply:
Either is fine by me too. Maybe round robin might be slightly preferable because it avoids the 0 size case in the last rank?